core/src/membank.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Robin Linden <dev@robinlinden.eu>
+
+#pragma once
+
+#include "core/imembank.h"
+
+#include <array>
+#include <cstdint>
+#include <utility>
+
+namespace n_e_s::core {
+
+template <uint16_t StartAddr,
+        uint16_t Size = 1u,
+        uint16_t EndAddr = StartAddr + (Size - 1u)>
+class MemBank : public IMemBank {
+public:
+    static_assert(Size > 0u, "Size must be greater than zero");
+    static_assert(StartAddr <= EndAddr, "Start addr greater than end addr");
+
+    MemBank() : bank_() {}
+
+    bool is_address_in_range(uint16_t addr) const override {
+        return addr >= StartAddr && addr <= EndAddr;
+    }
+
+    uint8_t read_byte(uint16_t addr) const override {
+        return *get_location(addr);
+    }
+
+    uint16_t read_word(uint16_t addr) const override {
+        return read_byte(addr) | read_byte(addr + 1) << 8;
+    }
+
+    void write_byte(uint16_t addr, uint8_t byte) override {
+        *get_location(addr) = byte;
+    }
+
+    void write_word(uint16_t addr, uint16_t word) override {
+        write_byte(addr, word & 0xFF);
+        write_byte(addr + 1, word >> 8);
+    }
+
+private:
+    uint8_t *get_location(uint16_t addr) {
+        return const_cast<uint8_t *>(std::as_const(*this).get_location(addr));
```

Comment:
I saw a guideline on cpp core guidelines regarding const_cast: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es50-dont-cast-away-const

They suggest using a templated helper function instead. What du you guys think?

Reply:
I don't mind either method. I think this way is fine because it requires us to have a mutable object before calling anything, but I think the template method thing is fine too.

Reply:
Ok if we have it as it is for now? We can consider changing it later.

Reply:
Sure!